docs/DeveloperGuide.md

```diff
@@ -643,80 +643,183 @@ otherwise)
-5. AnimalCatalog deletes the animal and displays the success message.
-   Use case ends.
+1. User requests to delete a task from the task list of an animal by specifying the animal's index and the task index.
+2. AnimalCatalog validates the animal and task indices and locates the specified task within the animal's task list.
+3. AnimalCatalog deletes the task from the animal's task list and displays a success message.
+    - Use case ends.
 
 **Extensions:**
-3a. AnimalCatalog detects that the given index is out of range.
-3a1. AnimalCatalog displays a failure message indicating an invalid index.
-Use case resumes from step 2.
+- 2a. The specified animal index is invalid.
+    - 2a1. AnimalCatalog displays an error message indicating an invalid animal index.
+    - Use case ends.
+- 2b. The specified task index is invalid.
+    - 2b1. AnimalCatalog displays an error message indicating an invalid task index.
+    - Use case ends.
 
-*a. At any time, User chooses to cancel the deletion.
-*a1. AnimalCatalog requests to confirm the cancellation.
-*a2. User confirms the cancellation.
-Use case ends.
+---
 
-**Use Case: UC05 - View Animal Details**
+**Use Case: UC09 - Mark Task as Done in Animal's Task List**
 
 **MSS:**
-1. User chooses to view details of animal.
-2. AnimalDetailView is updated with the selected animal.
-Use case ends.
+1. User requests to mark a task as done in the task list of an animal by specifying the animal's index and the task index(es).
+2. AnimalCatalog validates the animal and task indices and locates the specified task(s) within the animal's task list.
+3. AnimalCatalog marks the task(s) as done and displays a success message.
+    - Use case ends.
+
+**Extensions:**
+- 2a. The specified animal index is invalid.
+    - 2a1. AnimalCatalog displays an error message indicating an invalid animal index.
+    - Use case ends.
+- 2b. The specified task index(es) are invalid or exceed the number of tasks in the task list.
+    - 2b1. AnimalCatalog displays an error message indicating invalid or excessive task index(es).
+    - Use case ends.
+- 2c. One or more of the specified task index(es) do not correspond to an existing task.
+    - 2c1. AnimalCatalog displays an error message indicating the task does not exist.
+    - Use case ends.
+
+---
 
-*a. At any time, User enters a valid animal-specific command.
-*a1. AnimalDetailView is updated with the specified animal in the animal-specific command.
-Use case ends.
+**Use Case: UC10 - Unmark Task as Uncompleted in Animal's Task List**
+
+**MSS:**
+1. User requests to unmark a task as uncompleted in the task list of an animal by specifying the animal's index and the task index(es).
+2. AnimalCatalog validates the animal and task indices and locates the specified task(s) within the animal's task list.
+3. AnimalCatalog unmarks the task(s) as uncompleted and displays a success message.
+    - Use case ends.
+
+**Extensions:**
+- 2a. The specified animal index is invalid.
```

Comment:
All these extensions can be moved to 1a,1b,1c which makes more sense. Since its detecting an error after the user inputs to unmark a test.